Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js

```diff
@@ -80,18 +80,30 @@ function logError() {
 
 function detectMocha(projectFolder) {
   try {
-        var node_modulesFolder = projectFolder;
-        var mochaJsonPath = path.join(node_modulesFolder, 'test', 'mocha.json');
-        if (fs.existsSync(mochaJsonPath)) {
-          var opt = require(mochaJsonPath);
-          if (opt && opt.path) {
-            node_modulesFolder = path.resolve(projectFolder, opt.path);
-          }
+    var node_modulesFolder = projectFolder;
+
+    var nodePath = process.env.NODE_PATH;
+    if (nodePath && nodePath.length > 0) {
+      // get mocha location from NODE_PATH env 
+      var ndex = nodePath.search(new RegExp(/node_modules$/i));
```

Comment:
In general, don't wrap regexp literals in `new RegExp`.

But, just to be safe, make sure to use `path.sep` and a capture group to extract the path components:

```js
new RegExp('(.+?)' + path.sep + 'node_modules', 'i');
```